plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -348,11 +347,7 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
     @Override
     protected Optional<BiFunction<String, Long, String>> limitFunction()
     {
-        return Optional.of((sql, limit) -> {
-            String start = "SELECT ";
-            checkArgument(sql.startsWith(start));
-            return "SELECT TOP " + limit + " " + sql.substring(start.length());
-        });
+        return Optional.of((sql, limit) -> format("SELECT TOP %s * FROM (%s) o", limit, sql));
```

Comment:
nit : Since limit is a long , so can we provide proper format specifier for the same ?

Reply:
i use (we use?) `%s` for simplicity, unless i want to have type-specific formatting.